src/components/mobile/MobileProvider.tsx

```diff
@@ -53,14 +57,29 @@ export function MobileProvider({
         document.body.classList.toggle(rootMobileClassName, mobile);
     }, [rootMobileClassName, mobile]);
 
+    const touch = React.useMemo(() => isTouchScreen(), [width]);
```

Comment:
How does this depend on width?

Reply:
If we change device type from devtools, navigator can update context, for example if I toggle from desktop version to mobile, `navigator.maxTouchPoints` toggling value between 1 and 0

Reply:
We discuses this theme in more details [here](https://github.com/gravity-ui/uikit/pull/1764#discussion_r1728742337).
This case is same, devtools may change `navigator.maxTouchPoints` value, but we can't subscribe on this value directly. I deleted width observing from here because it's wrong direction for observing, unfortunately I didn't find how to subscribe on property changes of `isTouchScreen` method

Reply:
We discussed with @amje and @korvin89 should we use `MobileProvider` for inner statement, which uncontrolled by user, and decided to get rid of this and make special helper